File: spring-boot-sample/src/main/java/com/day1co/cornerstone/task/sample/SampleTask.java
package com.day1co.cornerstone.task.sample;

import com.day1co.cornerstone.commons.Task;
import com.day1co.cornerstone.commons.TaskException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class SampleTask implements Task<SampleMessage> {
    private static final Logger LOGGER = LoggerFactory.getLogger(SampleTask.class);
    private final SampleService sampleService;

    public SampleTask(SampleService watermark) {
        this.sampleService = watermark;
    }

    @Override
    public Class<SampleMessage> getMessageClass() {
        return SampleMessage.class;
    }

    @Override
    public void execute(SampleMessage message) {
        if (LOGGER.isDebugEnabled()) {
            LOGGER.debug("message: {}", message);
        }
        try {
            sampleService.doSomething();
        } catch (Throwable t) {
            throw new TaskException(t);
        }
    }

}

File: nodejs/sample-server/src/sample-task.ts
import { LoggerFactory } from '@day1co/pebbles';
import type { Task } from '@day1co/cornerstone-commons';

import { SampleService } from './sample-service';

const logger = LoggerFactory.getLogger('cornerstone-nodejs-sample:sample-task');

export class SampleTask implements Task<string> {
  constructor(readonly sampleService: SampleService) {}

  async execute(message: string): Promise<void> {
    logger.info(`message:${message}`);
    try {
      this.sampleService.doSomething(message);
    } catch (e) {
      throw e;
    }
  }
}
